### PR TITLE
DBC22-1183: sort cam list based on spatial route linestrings

### PR DESCRIPTION
**Pull request by ray-oxd**

### CODE REVIEW

File: src/backend/apps/webcam/tasks.py [hunk 1]
```diff
@@ -139,3 +150,71 @@ def update_webcam_image(webcam):
...
+
+# Camera ordering tasks
+def add_order_to_camera_group(key, cams):
+    # Only process if reference linestring exists
+    route_geometry = RouteGeometry.objects.filter(id=key).first()
+    if route_geometry:
+        # Combine all coordinates into a single linestring for sorting
+        all_coords = ()
+        for coord in route_geometry.routes.coords:
+            all_coords += coord
```
Comment (fatbird) [inline src/backend/apps/webcam/tasks.py:163]: How is all_coords necessary or different than route_geometry.routes.coords?  Why not use the latter to construct the LineString?

File: src/backend/apps/webcam/tasks.py [hunk 2]
```diff
@@ -139,3 +150,71 @@ def update_webcam_image(webcam):
...
+
+    # Group by route name and add order to cams in each group
+    for route_name, cam_group in groupby(query, lambda x: x.highway if x.highway != '0' else x.highway_description):
+        add_order_to_camera_group(route_name, list(cam_group))
+
+    # Rebuild cache
+    WebcamAPI().set_list_data()
+
+
+def build_route_geometries():
```
Comment (fatbird) [inline src/backend/apps/webcam/tasks.py:197]: Overall, we need an explanation in code of why we're doing this (meaning getting the route, storing it, and then using it to order cameras).  A docstring for this function would be a good place to outline the intended purpose (along with the JIRA ticket number for the story); the highway coords structure should then be commented to say "used by build_route_geometries".

**Review by fatbird — CHANGES_REQUESTED** [review]
See my comments for requested changes.

File: src/backend/apps/webcam/tasks.py (hunk 1 above)
Comment (ray-oxd) [reply src/backend/apps/webcam/tasks.py]: fixed logic via amend commit. This was supposed to combine coords of each route in route_geometry.routes (for highway 3A as I mentioned yesterday).

File: src/backend/apps/webcam/tasks.py (hunk 2 above)
Comment (ray-oxd) [reply src/backend/apps/webcam/tasks.py:203]: Updated docstrings and comments via amend commit.

**Review by fatbird — APPROVED** [review]

**Pull request merged**